### PR TITLE
Relocate parts of makeImmutable to lib/util.js

### PR DESCRIPTION
**Pull request by jdmarshall**
This PR builds on top of #864 

The second commit relocates all but the selective immutability code and puts it into Util. 

A subsequent PR will close a substantial bug in how makeImmutable works. IMO, #865 is a bug, so it will not receive a warning or a year grace period.

## Summary by CodeRabbit

* **New Features**
  * Added comprehensive utility method for object immutability with recursive support for nested objects and arrays.
  * Introduced method for selective property-level immutability.

* **Deprecations**
  * Existing immutability method marked as deprecated; users should migrate to new utility method.

* **Tests**
  * Added test coverage for immutability functionality including nested structures and edge cases.

### CODE REVIEW

File: lib/util.js [hunk 1]
```diff
@@ -113,6 +115,98 @@ class Util {
...
+            configurable: false
+          });
+        }
+      }
+    }
+
+    Object.preventExtensions(object)
+
+    return object;
+  }
```
[the remark this answers is not in the document]
Comment (jdmarshall) [reply lib/util.js:210]: Hallucination. Tests updated to prove.

File: lib/config.js [hunk 2]
```diff
@@ -55,402 +55,589 @@ let checkMutability = true;      // Check for mutability/immutability on first g
...
+        }
+      }
+    }
+
+    // https://github.com/lorenwest/node-config/issues/505
+    // https://github.com/lorenwest/node-config/issues/865
+    Object.preventExtensions(object)
+
+    return object;
+  }
```
[the remark this answers is not in the document]
Comment (jdmarshall) [reply lib/config.js:421]: I did extract the shared reusable logic. Maybe you should have adjusted your determination based on the fact that the old function is deprecated.

And while your second issue is technically true, it's not resulting in errors (and is a really old bug)

File: test/0-util.js [hunk 3]
```diff
@@ -115,6 +115,77 @@ describe('Tests for util functions', function () {
...
+    it('prevents addition of new fields', function() {
+      util.makeImmutable(object);
+
+      assert.throws(() => object.subObject.newField = "setToThis",
+        /Can not add runtime configuration property: "newField"/);
+    });
+
+    // This is the same bug #865
+    // it('objects in arrays should be immutable', function () {
+    //   util.makeImmutable(object);
```
[the remark this answers is not in the document]
Comment (jdmarshall) [reply test/0-util.js:164]: Added as an addendum to #865

File: lib/config.js [hunk 4]
```diff
@@ -55,402 +55,589 @@ let checkMutability = true;      // Check for mutability/immutability on first g
...
+      return toObject;
+    }
+
+    // Adding Config.prototype methods directly to toObject as hidden properties
+    // because adding to toObject.__proto__ exposes the function in toObject
+    for (const fnName of ['get', 'has', 'util', LOAD_SYMBOL]) {
+      if (!toObject[fnName]) {
+        Util.makeHidden(toObject, fnName, this.#config[fnName]);
       }
+    }
```
[the remark this answers is not in the document]
Comment (jdmarshall) [reply lib/config.js:546]: This is a known issue.  And a bug that also existed in the old code.

[the remark this answers is not in the document]
Comment (jdmarshall) [reply lib/config.js:546]: Additional commentary added to #795

File: lib/util.js [hunk 5]
```diff
@@ -116,6 +117,98 @@ class Util {
...
+        // Ensure object items of this array are also immutable.
+        for (let item of value) {
+          if (this.isObject(item) || Array.isArray(item)) {
+            this.makeImmutable(item);
+          }
+        }
+
+        Object.defineProperty(object, propertyName, {
+          value: Object.freeze(value)
+        });
```
[the remark this answers is not in the document]
Comment (jdmarshall) [reply lib/util.js:163]: This was already added to #865 as discussed before
